docs_user/modules/openstack-tls_everywhere.adoc

```diff
@@ -0,0 +1,211 @@
+* The issuer. The cert-manager operator makes use of a self-signed issuer to issue both the CA certificate itself and other certificates.
+* The CA certificate itself. We are specifically naming it as `rootca-internal`, which corresponds to the certificate authority leveraged by cert-manager to issue other certificates.
+
+Let's apply the following YAML file to create the corresponding CRs:
+
+[source,yaml]
+----
+oc apply -f - <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openstack
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: selfsigned-issuer
+  namespace: openstack
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: rootca-internal
+  namespace: openstack
+spec:
+  isCA: true
+  commonName: rootca-internal
+  secretName: rootca-internal
+  privateKey:
+    algorithm: RSA
+    size: $KEY_LENGTH
+  issuerRef:
+    name: selfsigned-issuer
+EOF
+----
+
+
+OK, you just created the CRs with the initial content. Now you will need to import the certificate and the key that you exported from FreeIPA. You can do this with the `oc patch` command. For the CA certificate (correspondeing to the `ca.crt` item on the Certificate, type:
+
+----
+oc patch secret rootca-internal -n openstack -p="{\"data\":{\"ca.crt\": \"`$IPA_SSH openssl pkcs12 -in /tmp/freeipa.p12 -passin file:/etc/pki/pki-tomcat/alias/pwdfile.txt -nokeys | openssl x509 | base64 -w 0`\"}}"
+----
+
+You will also need to update the `tls.crt` item. For the CA, this will receive the very same contents of `ca.crt`. The command just needs a small adjustment:
+
+----
+oc patch secret rootca-internal -n openstack -p="{\"data\":{\"tls.crt\": \"`$IPA_SSH openssl pkcs12 -in /tmp/freeipa.p12 -passin file:/etc/pki/pki-tomcat/alias/pwdfile.txt -nokeys | openssl x509 | base64 -w 0`\"}}"
+----
+
+Finally, update the key:
+
+----
+oc patch secret rootca-internal -n openstack -p="{\"data\":{\"tls.key\": \"`$IPA_SSH openssl pkcs12 -in /tmp/freeipa.p12 -passin file:/etc/pki/pki-tomcat/alias/pwdfile.txt -nocerts -noenc | openssl rsa | base64 -w 0`\"}}"
+----
+
+Now create the cert-manager Issuer, referencing the created secret:
+
+[source, yaml]
+----
+oc apply -f - <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openstack
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: rootca-internal
+  namespace: openstack
+  labels:
+    osp-rootca-issuer-internal: ""
+spec:
+  ca:
+    secretName: rootca-internal
+EOF
+----
+
+*Note: Do note forget to delete the p12 files you created during the previous steps, like `/tmp/freeipa.p12`!*
+
+==== Checking the newly provisioned CRs
+
+You can check the created resources with the commands below:
+
+----
+oc get issuers -n openstack
+----
+
+----
+oc get secret rootca-internal -n openstack -o yaml
+----
+
+== Stop and disable certmonger, and remove the existing certificates
```

Comment:
Replace with:

== Stop and disable certmonger, and remove the tracking requests for the existing certificates